lib/team_hub/cross_referencer.rb

```diff
@@ -0,0 +1,133 @@
+# team_hub - Components for creating a team Hub using Jekyll
+#
+# Written in 2015 by Mike Bland (michael.bland@gsa.gov)
+# on behalf of the 18F team, part of the US General Services Administration:
+# https://18f.gsa.gov/
+#
+# To the extent possible under law, the author(s) have dedicated all copyright
+# and related and neighboring rights to this software to the public domain
+# worldwide. This software is distributed without any warranty.
+#
+# You should have received a copy of the CC0 Public Domain Dedication along
+# with this software. If not, see
+# <https://creativecommons.org/publicdomain/zero/1.0/>.
+#
+# @author Mike Bland (michael.bland@gsa.gov)
+
+module TeamHub
+
+  # Operations for creating cross-references between data objects
+  class CrossReferencer
+    # Creates an index of +collection+ items based on +key+.
+    #
+    # @param collection [Array<Hash>] collection from which to build an index
+    # @param key [String] property used to build the index
+    # @return [Hash]
+    def self.create_index(collection, key)
+      index = {}
+      collection.each {|i| (index[i[key]] ||= Array.new) << i if i[key]}
+      index
+    end
```

Comment:
I'll be kiss my... Still needed three lines, on account of `Hash.delete nil` returns the deleted object instead of the original `Hash`, but nice, thanks! Done in a new commit.